tests/bluetooth/df/connection_cte_req/src/test_cte_set_rx_params.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <stddef.h>
+#include <ztest.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <sys/byteorder.h>
+#include <host/hci_core.h>
+
+#include "common.h"
+#include "test_cte_set_rx_params.h"
+
+static uint16_t g_conn_handle;
+
+static uint8_t g_ant_ids[] = { 0x1, 0x2, 0x3, 0x4, 0x5 };
+
+static struct ut_bt_df_conn_cte_rx_params g_params;
+
+/* Macros delivering common values for unit tests */
+#define CONN_HANDLE_INVALID (CONFIG_BT_MAX_CONN + 1)
+#define ANTENNA_SWITCHING_SLOT_INVALID 0x3 /* BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US + 1 */
+
+/* Antenna switch pattern length is stored in 1 octet. If BT Core spec. extends the max value to
+ * UINT8_MAX expected failures may not be checked. If storage size is increased, tests shall be
+ * updated.
+ */
+BUILD_ASSERT(CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN < UINT8_MAX, "Can't test expected failures "
+								   "for switch pattern length "
+								   "longer or equal to UINT8_MAX.");
+#define SWITCH_PATTERN_LEN_TOO_LONG (CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN + 1)
+
+BUILD_ASSERT(BT_HCI_LE_SWITCH_PATTERN_LEN_MIN > 0x0, "Can't test expected failures for switch "
+						     "pattern length smaller than zero.");
+#define SWITCH_PATTERN_LEN_TOO_SHORT (BT_HCI_LE_SWITCH_PATTERN_LEN_MIN - 1)
+
+int send_set_conn_cte_rx_params(uint16_t conn_handle, struct ut_bt_df_conn_cte_rx_params *params,
+				bool enable)
+{
+	struct bt_hci_cp_le_set_conn_cte_rx_params *cp;
+	struct net_buf *buf;
+
+	uint8_t ant_ids_num = (params != NULL ? params->switch_pattern_len : 0);
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_CONN_CTE_RX_PARAMS, sizeof(*cp) + ant_ids_num);
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(conn_handle);
+	cp->sampling_enable = enable ? 1 : 0;
+
+	if (params != NULL) {
+		cp->slot_durations = params->slot_durations;
+
+		if (params->switch_pattern_len) {
+			uint8_t *dest_ant_ids = net_buf_add(buf, params->switch_pattern_len);
+
+			if (params->ant_ids) {
+				memcpy(dest_ant_ids, params->ant_ids, params->switch_pattern_len);
```

Comment:
```suggestion
				memcpy(dest_ant_ids, params->ant_ids, ant_ids_num);
```

Reply:
```suggestion
				(void)memcpy(dest_ant_ids, params->ant_ids, params->switch_pattern_len);
```

Reply:
Applied first suggestion. Why add cast to `void` before `memcpy`?

Reply:
The same reason why you cast `(void)memset`: Because `memcpy` returns a pointer that you don't use. 

I can't remember if there's any strict requirement for handling all return values in Zephyr (or if there ever will be), but if you do it for `memset` then you should also do it for `memcpy` if not just for consistency.

Reply:
Added missing cast to `(void)`.